NXP S32ZE update to HAL drivers version 2.0.1

tunguyen4585 opened the pull request:
NXP S32ZE update to HAL drivers version 2.0.1

manuargue reviewed (changes requested):
pls split commits for adc, eth and pwm and add details on each commit on why the changes are needed

tunguyen4585 committed:
drivers: adc: update new macro name in RTD 2.0.1

FEATURE_ADC_MAX_CHN_COUNT changed to ADC_SAR_IP_MAX_CHN_COUNT
in RTD 2.0.1

Signed-off-by: Tu Nguyen Van <tu.nguyenvan@nxp.com>

tunguyen4585 committed:
drivers: pwm: update variable and marco name in RTD 2.0.1

eMios_Icu_Ip_IndexInChState is changed to
eMios_Icu_Ip_u8IndexInChState and EMIOS_ICU_BUS_F
is removed in RTD 2.0.1

Signed-off-by: Tu Nguyen Van <tu.nguyenvan@nxp.com>

tunguyen4585 committed:
drivers: ethernet: adapt code with new RTD 2.0.1

update members in struct Netc_Eth_Ip_StationInterfaceConfigType
in RTD 2.0.1 for both psi and vsi driver code.

Netc_Eth_Ip_VsiToPsiMsgType need to relocate nocache section
in RTD 2.0.1

Signed-off-by: Tu Nguyen Van <tu.nguyenvan@nxp.com>
